src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,165 @@
+----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Bipartite.AdjacencyMap
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for
+-- the motivation behind the library, the underlying theory, and
+-- implementation details.
+--
+-- This module defines the 'Bipartite.AdjacencyMap' for bipartite graphs data
+-- type and basic associated functions.
+----------------------------------------------------------------------------
+module Algebra.Graph.Bipartite.AdjacencyMap (
+    -- * Data structure
+    AdjacencyMap, leftToRight, rightToLeft,
+
+    -- * Conversion functions
+    fromAdjacencyMap, toAdjacencyMap, fromGraph,
+
+    -- * Graph properties
+    hasEdge,
+
+    -- * Miscellaneous
+    consistent,
+    ) where
+
+import Data.Either (lefts, rights)
+import Data.List   (sort)
+import Data.Tuple  (swap)
+
+import qualified Algebra.Graph              as G
+import qualified Algebra.Graph.AdjacencyMap as AM
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Set        as Set
+
+-- | The 'Bipartite.AdjacencyMap' data type represents an __undirected__
+-- bipartite graph by two maps of vertices into its adjacency sets. The two
+-- type arguments for this datatype stand for the identifiers of the vertices
+-- of each part.
+--
+--  __Note:__ even if the identifiers and their types for two vertices of
+-- different parts are equal, these vertices are considered to be different.
+-- See the examples for more details.
+data AdjacencyMap a b = BAM {
+    -- | The /adjacency map/ of the left part of the graph: each vertex is
+    -- associated with a set of its neighbours. Complexity: /O(1)/ time and
+    -- memory
+    --
+    -- @
+    -- leftToRight ('fromGraph' 'Algebra.Graph.empty')                                          == Map.'Map.empty'
```

Comment:
I think it may be better to follow the naming convention in the rest of the library, where such functions are typically named `adjacencyMap`. In this case we could call them `leftAdjacencyMap` and `rightAdjacencyMap`.